src/env.c

```diff
@@ -25,38 +25,43 @@ extern "C"
 #include "rcutils/error_handling.h"
 
 bool
-rcutils_set_env(const char * env_name, const char * env_value)
+rcutils_set_env2(const char * env_name, const char * env_value, bool overwrite)
 {
   RCUTILS_CAN_RETURN_WITH_ERROR_OF(false);
 
   RCUTILS_CHECK_FOR_NULL_WITH_MSG(
     env_name, "env_name is null", return false);
 
+  if (getenv(env_name) != NULL) {
+    if (overwrite == 0) {
+      return true;
+    }
+  }
+
+  int set_ret;
 #ifdef _WIN32
   if (NULL == env_value) {
```

Comment:
There is a segfault here

```cpp
ASSERT_TRUE(rcutils_set_env2("NEW_ENV_VAR", nullptr, true));
```

when `env_value` inside this function is null, I think we can move this check outside of the `ifdef`